src/executors/PromiseExecutor.js

```diff
@@ -1,7 +1,36 @@
 import ExecutorAbstract from '../ExecutorAbstract';
 
 class PromiseExecutor extends ExecutorAbstract {
+  //-----------------------------------
+  // Constructor
+  //-----------------------------------
+  /**
+   * Constructor for a new PromiseExecutor
+   *
+   * @param thread The parent thread that owns this executor.
+   * @param promise The promise to execute
+   */
+  constructor(thread, promise) {
+    super(thread);
 
+    this.promise = promise;
+  }
+
+  //-----------------------------------
+  // Methods
+  //-----------------------------------
+  /**
+   * Internal execution method called by Thread only.
+   *
+   * @param doneHandler The handler to call when done() is called.
+   * @param failHandler The handler to call when fail() is called;
+   * @private
+   */
+  _execute(doneHandler, failHandler) {
+    super._execute(doneHandler, failHandler);
+
+    this.promise.then(this.done).catch(this.fail);
```

Comment:
Promises have a result that is a parameter. I'm not sure that this is going to work or else that result will be passed to done and the first parameter to done is not a result. Same goes for fail.

We also need to provide the functionality that a person can explicitly set the property they want set with the promise result and error.

Also, does the ExecutorFactory automagically create a PromiseExecutor when you provide a promise?